File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/CS/ServerRevokeAccessTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

#if !SILVERLIGHT
using System;
using Db4oUnit;
using Db4oUnit.Extensions;
using Db4oUnit.Extensions.Fixtures;
using Db4objects.Db4o;
using Db4objects.Db4o.CS;
using Db4objects.Db4o.CS.Internal.Config;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Tests.Common.CS;

namespace Db4objects.Db4o.Tests.Common.CS
{
	public class ServerRevokeAccessTestCase : Db4oClientServerTestCase, IOptOutAllButNetworkingCS
	{
		private static readonly string ServerHostname = "127.0.0.1";

		public static void Main(string[] args)
		{
			new ServerRevokeAccessTestCase().RunAll();
		}

		#if !CF
		/// <exception cref="System.IO.IOException"></exception>
		public virtual void Test()
		{
			string user = "hohohi";
			string password = "hohoho";
			IObjectServer server = ClientServerFixture().Server();
			server.GrantAccess(user, password);
			IObjectContainer con = OpenClient(user, password);
			Assert.IsNotNull(con);
			con.Close();
			server.Ext().RevokeAccess(user);
			Assert.Expect(typeof(Exception), new _ICodeBlock_40(this, user, password));
		}
		#endif // !CF

		private sealed class _ICodeBlock_40 : ICodeBlock
		{
			public _ICodeBlock_40(ServerRevokeAccessTestCase _enclosing, string user, string 
				password)
			{
				this._enclosing = _enclosing;
				this.user = user;
				this.password = password;
			}

			/// <exception cref="System.Exception"></exception>
			public void Run()
			{
				this._enclosing.OpenClient(user, password);
			}

			private readonly ServerRevokeAccessTestCase _enclosing;

			private readonly string user;

			private readonly string password;
		}

		private IObjectContainer OpenClient(string user, string password)
		{
			return Db4oClientServer.OpenClient(Db4oClientServerLegacyConfigurationBridge.AsClientConfiguration
				(Config()), ServerHostname, ClientServerFixture().ServerPort(), user, password);
		}

		private IConfiguration Config()
		{
			return ClientServerFixture().Config();
		}
	}
}
#endif // !SILVERLIGHT

File: db4oj.tests/src/com/db4o/db4ounit/common/cs/ServerRevokeAccessTestCase.java
/* Copyright (C) 2004   Versant Inc.   http://www.db4o.com */

package com.db4o.db4ounit.common.cs;

import java.io.*;

import com.db4o.*;
import com.db4o.config.*;
import com.db4o.cs.internal.config.*;

import db4ounit.*;
import db4ounit.extensions.*;
import db4ounit.extensions.fixtures.*;

public class ServerRevokeAccessTestCase
	extends Db4oClientServerTestCase
	implements OptOutAllButNetworkingCS {

	private static final String SERVER_HOSTNAME = "127.0.0.1";

	public static void main(String[] args) {
		new ServerRevokeAccessTestCase().runAll();
	}

	/**
	 * @sharpen.if !CF
	 */
	public void test() throws IOException {
		final String user = "hohohi";
		final String password = "hohoho";
		ObjectServer server = clientServerFixture().server();
		server.grantAccess(user, password);

		ObjectContainer con = openClient(user, password);
		Assert.isNotNull(con);
		con.close();

		server.ext().revokeAccess(user);

		Assert.expect(Exception.class, new CodeBlock() {
			public void run() throws Throwable {
				openClient(user, password);
			}
		});
	}

	private ObjectContainer openClient(final String user, final String password) {
		return com.db4o.cs.Db4oClientServer.openClient(Db4oClientServerLegacyConfigurationBridge.asClientConfiguration(config()), SERVER_HOSTNAME,
				clientServerFixture().serverPort(), user, password);
	}

	private Configuration config() {
	    return clientServerFixture().config();
    }
}
